packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +105,86 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// This class stores the information of an active selection under
+/// a [Selectable] or [SelectionHandler], including the start and
+/// end positions of a selection relative to the content, as well
+/// as an identifier to identify the owner of the content, any
+/// children selection ranges that may be contained within the active
+/// selection, and the content itself.
+///
+/// [SelectionArea] and [SelectableRegion] provide access to this
+/// information through the [SelectionListener] passed to their
+/// [SelectionListener.onSelectionChanged] member.
+///
+/// See also:
+///
+///   * [SelectionListener], which provides the [SelectedContentRange]s representing
+///     the active selection for a given subtree contained under a [SelectionArea] 
+///     or [SelectableRegion].
+abstract class SelectedContentRange<T extends Object> {
+  /// Creates a range for the content of a [Selectable] or [SelectionHandler].
+  SelectedContentRange({
+    this.selectableId,
```

Comment:
Also maybe provide an example of how to use this selectableId in a practical situation? The way you get a selectableId from Text seems like it might be beyond the ability most users if they wanted to set something similar up for their own class.

Reply:
Do you mean if a user makes a wrapper around a `Text` widget? or do you mean if a user is making their own custom widget selectable?

Reply:
I added this to the existing example selectable_region.0. It's pretty basic though since that example's `MySelectableAdapter` only selects all or nothing.